jwst/assign_wcs/nirspec.py

```diff
@@ -171,7 +172,14 @@ def ifu(input_model, reference_files, slit_y_range=[-.55, .55]):
     detector = input_model.meta.instrument.detector
     grating = input_model.meta.instrument.grating
     filter = input_model.meta.instrument.filter
-
+    # Check for ifu reference files
+    if reference_files['ifufore'] is None and \
+       reference_files['ifuslicer'] is None and \
+       reference_files['ifupost'] is None:
+        # No ifu reference files, won't be able to create pipeline
+        log_message = 'No ifufore, ifuslicer or ifupost reference files'
+        log.critical(log_message)
+        raise RuntimeError(log_message)
```

Comment:
Does the RuntimeError get trapped up in the pipeline module, so that it can shut down cleanly?

Reply:
calwebb_spec2 traps all exceptions in assign_wcs, so it should look like any other assign_wcs problem

Reply:
I've been thinking forever (and never did anything about it) that something like this is needed in the assign_wcs_step.py module when the reference files are pulled from CRDS. assign_wcs requires and will not run without reference files. Some of them apply to all modes but it's nice that you checked for the ifu specific ones here.